Simple wrapper script + hierarchical Makefiles.

NGenetzky opened the pull request:
Very simple wrapper than only determines which executable to call. Also moved the Makefile organization to a hierarchical organization

dhill-usgs commented on scripts/cfmask.py:
```diff
@@ -0,0 +1,125 @@
+            return False
+
+
+class EXECUTE_ERROR(Exception):
```
classes are not all caps

NGenetzky replied:
It was originally a constant. Fixed.
```class ExecuteError(Exception):```

rdilley-usgs commented on scripts/Makefile:
```diff
@@ -0,0 +1,14 @@
+install:
+	install -d $(PREFIX)/bin
+	install -m 775 cfmask.py $(PREFIX)/bin
+clean:
```
Add a space before clean:

NGenetzky replied:
Fixed.

rdilley-usgs commented on scripts/cfmask.py:
```diff
@@ -0,0 +1,125 @@
+        if len(output) > 0:
+            logger.info("\n{0}".format(output))
+    except EXECUTE_ERROR:
+        logger.exception(('Error running {0}.'
```
Don't need the extra parens wrapping the string for the format command.  (('Error   ....  ).format

NGenetzky replied:
Fixed.
```
logger.exception('Error running {0}.'
                         'Processing will terminate.'
                         .format(os.path.basename(__file__)))
```

rdilley-usgs commented on scripts/cfmask.py:
```diff
@@ -0,0 +1,125 @@
+            raise EXECUTE_ERROR(message)
+
+        if status != 0:
+            message = "Application failed to execute [%s]" % cmd_string
```
Switch this to use ''.format, as well as single quotes.

rdilley-usgs commented on scripts/cfmask.py:
```diff
@@ -0,0 +1,125 @@
+        (status, output) = commands.getstatusoutput(cmd_string)
+
+        if status < 0:
+            message = "Application terminated by signal [%s]" % cmd_string
```
Switch this to use ''.format, as well as single quotes.

rdilley-usgs commented on scripts/cfmask.py:
```diff
@@ -0,0 +1,125 @@
+            raise EXECUTE_ERROR(message)
+
+        if os.WEXITSTATUS(status) != 0:
+            message = ("Application [%s] returned error code [%d]"
```
Switch this to use ''.format, as well as single quotes.

rdilley-usgs commented on scripts/cfmask.py:
```diff
@@ -0,0 +1,125 @@
+        output = execute_cmd(cmd_string)
+
+        if len(output) > 0:
+            logger.info("\n{0}".format(output))
```
Use single quotes

rdilley-usgs commented on scripts/cfmask.py:
```diff
@@ -0,0 +1,125 @@
+    cmd.extend(sys.argv[1:])  # Pass all arguments through
+    cmd_string = ' '.join(cmd)
+    try:
+        logger.info('>>'+cmd_string)
```
Use ''.format here instead of +

NGenetzky commented:
Fixed all single quotes. Also changed all %s string modification to str.format().
```
if status < 0:
            message = ('Application terminated by signal [{0}]'
                       .format(cmd_string))
            if len(output) > 0:
                message = ' Stdout/Stderr is: '.join([message, output])
            raise ExecuteError(message)

        if status != 0:
            message = 'Application failed to execute [{0}]'.format(cmd_string)
            if len(output) > 0:
                message = ' Stdout/Stderr is: '.join([message, output])
            raise ExecuteError(message)

        if os.WEXITSTATUS(status) != 0:
            message = ('Application [{0}] returned error code [{1}]'
                       .format(cmd_string, os.WEXITSTATUS(status)))
            if len(output) > 0:
                message = ' Stdout/Stderr is: '.join([message, output])
            raise ExecuteError(message)
```
